### PR TITLE
Refactor build process + include css

### PR DESCRIPTION
**Pull request by krissalvador27**
- refactor build process with webpack
- include css style in build
- todo update readme.md to include import styles.css in how to section

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,19 +15,25 @@
...
     "docs": "npm run make-docs && npm run build-docs",
-    "build-lib": "npm run clean && babel src --out-dir ./ --source-maps",
-    "build-docs": "BABEL_ENV=production webpack --config webpack.config.build.js",
+    "build-lib":
+      "BABEL_ENV=production webpack --config webpack.config.build-lib.js",
+    "build-docs":
+      "BABEL_ENV=production webpack --config webpack.config.build-docs.js",
     "make-docs": "node scripts/makeDocs.js",
-    "clean": "node scripts/clean.js",
+    "clean": "rm -rf ./dist",
```
Comment (joshbDev) [inline package.json]: does this still clean the files that get dumped on the root?

File: package.json [hunk 2]
```diff
@@ -20,19 +15,25 @@
...
     "dev": "webpack-dev-server --config webpack.config.base.js",
     "docs": "npm run make-docs && npm run build-docs",
-    "build-lib": "npm run clean && babel src --out-dir ./ --source-maps",
-    "build-docs": "BABEL_ENV=production webpack --config webpack.config.build.js",
+    "build-lib":
+      "BABEL_ENV=production webpack --config webpack.config.build-lib.js",
+    "build-docs":
+      "BABEL_ENV=production webpack --config webpack.config.build-docs.js",
     "make-docs": "node scripts/makeDocs.js",
-    "clean": "node scripts/clean.js",
```
Comment (joshbDev) [inline package.json]: are you able to remove `scripts/clean.js`?

File: styles/charts.less [hunk 3]
```diff
@@ -0,0 +1,147 @@
+.chart-tick {
```
Comment (joshbDev) [inline styles/charts.less]: nitpick: in the future would you mind submitting two PR's, one for Webpack and one for styles? the merging of the two concerns is throwing me off a bit.

**Comment by dandelany** [comment]
A few thoughts:

- Library code really shouldn't be built into a bundle, but should remain as separate files, for the sake of tree-shaking and only importing what you need
- For this reason, I don't think webpack is really suitable/necessary for building the library code, it really only needs to be transpiled which can be done directly with command-line `babel`. It makes sense for the docs, which is a self-contained web app, but not for the library itself.
- I also don't think the library code should be built into `dist` - we did this originally before changing to build into root, since it was more confusing/annoying to have to import eg. `reactochart/dist/Thing` rather than just `reactochart/Thing`. Lodash and most other libraries build to root.

TL;DR I think this refactor is a bit too "baby-out-with-the-bathwater"... The styles should be included in the library, but I don't think that alone justifies redoing the rest of the build process, which was pretty mature. IMHO you could just add a `less` command line script for doing that part (something like `"build-styles": "less styles/charts.less charts.css"`)

**Comment by krissalvador27** [comment]
Yeah realizing now the bundle approach is definitely a step back. Thought it would be worthwhile to consolidate the doc and library build to use the same process while using webpack plugins to extract less files to single css file. Went down the rabbit hole with some webpack issues and ended up with this "baby-out-with-the-bathwater" PR. Closing in favor of the much simpler approach https://github.com/spotify/reactochart/pull/63
